core/src/main/java/org/opensearch/remote/metadata/client/impl/LocalClusterIndicesClient.java

```diff
@@ -193,6 +197,9 @@ protected CompletionStage<GetDataObjectResponse> innerGetDataObjectAsync(
         Executor executor,
         Boolean isMultiTenancyEnabled
     ) {
+        if (!Objects.isNull(request.cmkRoleArn())) {
+            throw new IllegalArgumentException("Local cluster client doesn't support cmk.");
+        }
```

Comment:
While you've refactored the location of this, the question still remains: why do we need to throw an exception here, rather than just silently ignoring the cmk field?

The main use case here is going to be in a plugin, where we may not know which client we're using in the code (it's a runtime configuration) so the code should probably add the cmk field always; but if it's on the local client it can be ignored.

How is this handled without forking the code to have a cmk and non-cmk version?

Reply:
Got your point. Considering the plugin usage, it's better to silently accept the CMK instead of throwing error. Already change the code.